### PR TITLE
RSDK-6344 Read partial config when getting cached TLS information

### PR DESCRIPTION
**Pull request by maximpertsov**
When RDK pulls a new cloud config file, it still tries to read TLS certificate data from the cached config if it exists. However, RDK validates the entire config file at this step and fails hard if there is any validation errors (aborting startup). 

Since RDK only needs TLS data at this point, it should not validate the cached config for anything beyond the relevant TLS fields.

### CODE REVIEW

File: config/reader.go [hunk 1]
```diff
@@ -85,6 +85,29 @@ func getCloudCacheFilePath(id string) string {
 	return filepath.Join(ViamDotDir, fmt.Sprintf("cached_cloud_config_%s.json", id))
 }
 
+type cachedTLSConfig struct {
+	Certificate string `json:"tls_certificate"`
+	PrivateKey  string `json:"tls_private_key"`
+}
+
+func readTLSFromCache(id string) (*cachedTLSConfig, error) {
```
Comment (cheukt) [inline config/reader.go:93]: could we use readFromCache and then only grab the TLS section out of it? code seems pretty similar

Comment (maximpertsov) [reply config/reader.go:93]: it looks like that function expects json with the full cloud config object? let me try though

**Review by cheukt — APPROVED** [review]
LGTM if it's been tested. minor nit would be to separate getTLSFromCache to a separate function so it's a little more readable

**Comment by maximpertsov** [comment]
> LGTM if it's been tested. minor nit would be to separate getTLSFromCache to a separate function so it's a little more readable

Yup, I tested "manually" by removing required fields from the cached config and loading RDK. It started up as expected.

**Comment by maximpertsov** [comment]
> > > minor nit would be to separate getTLSFromCache to a separate function so it's a little more readable
> > 
> > 
> > I agree, will make separate function
> 
> scratch that... it's a little awkward to do this since the flow has a few different outcomes (failing and exiting, failing and continuing, succeeding and setting the fields). I'll just update the comment to be a bit more clear about what we are doing here.

Ended up doing this after all: https://github.com/viamrobotics/rdk/pull/3620/commits/f8e724446f0999f867ab3c5a9298cbd0a2d04f7a - let me know if you think this is too much @cheukt

**Pull request merged**